lib/clouds/libcloud_common.py

```diff
@@ -1103,48 +1221,81 @@ def vmrunstate(self, obj_attr_list) :
 
-        except AWSException, obj :
-            _status = int(obj.error_code)
-            _fmsg = str(obj.error_message)
-
         except Exception, e :
             _status = 23
             _fmsg = str(e)
 
         finally :
-            if _status :
-                _msg = "Instance " + obj_attr_list["uuid"] + " could not have its "
-                _msg += "run state changed on " + self.description + " \""
-                _msg += obj_attr_list["cloud_name"] + "\" : " + _fmsg
-                cberr(_msg, True)
-                raise CldOpsException(_msg, _status)
-            else :
-                _msg = "Instance " + obj_attr_list["uuid"] + " successfully had its "
-                _msg += "run state changed on " + self.description + " \""
-                _msg += obj_attr_list["cloud_name"] + "\"."
-                cbdebug(_msg, True)
-                return _status, _msg
+            _status, _msg = self.common_messages("VM", obj_attr_list, "runstate altered", _status, _fmsg)
+            return _status, _msg
 
     @trace
-    def rotate_token(self, cloud_name) :
-        vmc_defaults = self.osci.get_object(cloud_name, "GLOBAL", False, "vmc_defaults", False)
-        credentials_lists = vmc_defaults["credentials"].split(",")
-        lock = self.lock(cloud_name, "VMC", "shared_access_token_counter", "credentials_list")
+    def vmmigrate(self, obj_attr_list) :
+        '''
+        TBD
+        '''
+        return 0, "NOT SUPPORTED"
 
-        assert(lock)
+    @trace
+    def vmresize(self, obj_attr_list) :
+        '''
+        TBD
+        '''
+        return 0, "NOT SUPPORTED"
 
-        current_token = 0 if "current_token" not in vmc_defaults else int(vmc_defaults["current_token"])
-        new_token = current_token
+    @trace
+    def imgdelete(self, obj_attr_list) :
+        '''
+        TBD
+        '''
+        try :
+            _status = 100
+            _image_instance = False
+            
+            _fmsg = "An error has occurred, but no error message was captured"
+            
+            self.common_messages("IMG", obj_attr_list, "deleting", 0, '')
 
-        if len(credentials_lists) > 1 :
-            new_token += 1
-            if new_token == len(credentials_lists) :
-                new_token = 0
+            _credentials_list = self.rotate_token(obj_attr_list["cloud_name"])
 
-        self.osci.update_object_attribute(cloud_name, "GLOBAL", "vmc_defaults", False, "current_token", str(new_token))
-        self.unlock(cloud_name, "VMC", "shared_access_token_counter", lock)
+            obj_attr_list["credentials_list"] = _credentials_list
+
+            _image_instance = self.get_images(obj_attr_list)
+                
+            if _image_instance :
+
+                _x = obj_attr_list["imageid1"] 
+                obj_attr_list["imageid1"] = _image_instance.id
+                
+                LibcloudCmds.catalogs.cbtool[_credentials_list].delete_image(_image_instance)
```

Comment:
So, I tested the imagedelete with this code, and it worked... I could see the newly created image   properly disappear from the Digital Ocean dashboard...